src/DepositHandler.sol

```diff
@@ -1,45 +1,58 @@
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {IDepositHandlerErrors} from "./interfaces/ICustomErrors.sol";
 
-contract DepositHandler {
+/**
+ * @title Deposit Handler contract.
+ * @author @ohMySol, @nynko, @ok567
+ * @dev Contract for managing users deposits for bootcamps.
+ * Implements both user part and admin part for deposit management.
+ * Apart from that contract allow to manage admins and bootcamps.
+ */
+contract DepositHandler is IDepositHandlerErrors, Pausable, Ownable {
+    IERC20 public immutable usdc; //at the moment assume that Encode using only USDC.
     address public admin;
-    IERC20 public usdcToken;
-    //mapping(address => uint256) public deposits; 
-    //mapping(address => bool) public bootcampCompleted;
-    mapping(address => depositInfo) public userDepositInfo;
-
     struct depositInfo{
         uint256 depositedAmount;
         bool bootcampCompleted; //this is set by an admin from encode (Centralised)
-
         bool yeild; //have not set this up yet
         bool multiSender; //have not set this up yet
-
     }
 
-
-    constructor(address _usdcToken) {
+    //mapping(address => uint256) public deposits; 
+    //mapping(address => bool) public bootcampCompleted;
+    mapping(address => depositInfo) public userDepositInfo;
+    
+    constructor(address _usdcToken) Ownable(msg.sender) {
+        usdc = IERC20(_usdcToken);
         admin = msg.sender;
-        usdcToken = IERC20(_usdcToken);
     }
 
-    // Allow users to deposit USDC
-    function deposit(uint256 amount) external { 
-        require(amount >= 249e18 && amount <= 251e18, "Deposit must be 250USDC"); //maybe try and not hardcode, try and get the deposit value from the bootcampmanager 
-        require(usdcToken.transferFrom(msg.sender, address(this), amount), "Transfer failed");
-        userDepositInfo[msg.sender].depositedAmount += amount;
-    }
-
-    // Mark a user as having completed the bootcamp (only admin)
-    function markBootcampComplete(address user) external {
-        require(msg.sender == admin, "Only admin can mark completion"); //this could be done with a modifier instead to save gas?
-        userDepositInfo[user].bootcampCompleted = true;
+    /*//////////////////////////////////////////////////
+                USER FUNCTIONS
+    /////////////////////////////////////////////////*/
+    /**
+     * @dev Allow user to deposit USDC '_amount' for a specific bootcamp. 
+     * Deposited amount will be locked inside this contract till the end of
+     * the bootcamp.
+     * @param _amount - USDC amount.
+     */
+    function deposit(uint256 _amount) external whenNotPaused { 
+        require(_amount >= 249e18 && _amount <= 251e18, "Deposit must be 250USDC"); //maybe try and not hardcode, try and get the deposit value from the bootcampmanager +
+        require(usdc.transferFrom(msg.sender, address(this), _amount), "Transfer failed");
+        userDepositInfo[msg.sender].depositedAmount += _amount;
```

Comment:
Note for when we will be improving gas: never put unchecked there haha

Reply:
100%, + we will improve and rewrite this function I guess.